client/pkg/fileutil/purge.go

```diff
@@ -63,6 +69,7 @@ func purgeFile(lg *zap.Logger, dirname string, suffix string, max uint, interval
 				f := filepath.Join(dirname, newfnames[0])
 				l, err := TryLockFile(f, os.O_WRONLY, PrivateFileMode)
 				if err != nil {
+					lg.Warn("failed to lock file", zap.String("file", f), zap.Error(err))
 					break
 				}
 				if err = os.Remove(f); err != nil {
```

Comment:
Should we also log error from here?

Reply:
Updated

Reply:
When etcd doesn't trigger one snapshot and doesn't free the wal file lock, and the wal file is more than 5(the default max wal file). This line will print every 30s(the default purge file time.)

Just like:

```
{"level":"warn","ts":"2024-08-29T06:35:26.029452Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:35:56.029743Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:36:26.029979Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:36:56.03078Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:37:26.031225Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:37:56.031624Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:38:26.031794Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:38:56.032833Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:39:26.033029Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:39:56.03381Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
{"level":"warn","ts":"2024-08-29T06:40:26.035535Z","caller":"fileutil/purge.go:80","msg":"failed to lock file","path":"/var/lib/etcd/data/member/wal/0000000000000000-0000000000000000.wal","error":"fileutil: file already locked"}
```

I am not sure whether we should do something to avoid this?